src/gitlab.js

```diff
@@ -19,16 +20,17 @@ export type VerdaccioGitlabAccessLevel =
 export type VerdaccioGitlabConfig = {
   url: string,
   authCache?: {
-    enabled?: boolean,
     ttl?: number
   },
   legacy_mode?: boolean,
+  access?: VerdaccioGitlabAccessLevel,
   publish?: VerdaccioGitlabAccessLevel
 };
 
 export type VerdaccioGitlabPackageAccess = PackageAccess & {
   name: string,
-  gitlab?: boolean
+  gitlab?: boolean,
+  gitlabUserGroupWhitelist?: any
```

Comment:
A space-separated list of gitlab groups. When defined, the user has to be member of any group to be able to access or publish, allowing truly private registries.